bundles/org.openhab.binding.easee/src/main/java/org/openhab/binding/easee/internal/connector/WebInterface.java

```diff
@@ -214,12 +210,12 @@ private synchronized void authenticate() {
          * periodically refreshed the access token.
          */
         private synchronized void refreshAccessToken() {
-            Date now = new Date();
+            Instant now = Instant.now();
             long expiryBuffer = TimeUnit.MINUTES.toMillis(WEB_REQUEST_TOKEN_EXPIRY_BUFFER_MINUTES);
             long maxAge = TimeUnit.MINUTES.toMillis(WEB_REQUEST_TOKEN_MAX_AGE_MINUTES);
 
-            if (tokenExpiry.getTime() - now.getTime() - expiryBuffer < 0
-                    || tokenRefreshDate.getTime() + maxAge < now.getTime()) {
+            if (tokenExpiry.toEpochMilli() - now.toEpochMilli() - expiryBuffer < 0
+                    || tokenRefreshDate.toEpochMilli() + maxAge < now.toEpochMilli()) {
```

Comment:
Now that you have Instants, this could be simplified to something like:

```java
if (now.isAfter(tokenExpiry.plusMinutes(WEB_REQUEST_TOKEN_EXPIRY_BUFFER_MINUTES))
        || now.isAfter(tokenRefreshDate.plusMinutes(WEB_REQUEST_TOKEN_MAX_AGE_MINUTES)))
{
```

(please double-check logic, point is that you don't need to convert everything to epoch and milliseconds)

Reply:
plusMinutes is not a member of Instant and also calculation is wrong. But based on your suggestion I changed it. :-)